frontend/test/unit/specs/components/v-search-results-title.spec.js

```diff
@@ -2,19 +2,76 @@ import { render } from "~~/test/unit/test-utils/render"
 
 import VSearchResultsTitle from "~/components/VSearchResultsTitle.vue"
 
+const DEFAULT_PROPS = {
+  size: "large",
+  searchTerm: "zack",
+  searchType: "all",
+  resultCounts: [
+    ["image", 10],
+    ["audio", 10],
+  ],
+}
+
 describe("VSearchResultsTitle", () => {
-  let options = {
+  const getOptions = ({ props, ...options } = {}) => ({
+    ...options,
     props: {
-      size: "large",
-    },
-    scopedSlots: {
-      default: () => "zack",
+      ...DEFAULT_PROPS,
+      ...props,
     },
-  }
+  })
 
   it("should render an h1 tag containing the correct text", async () => {
-    const { findByText } = render(VSearchResultsTitle, options)
-    const button = await findByText("zack")
-    expect(button.tagName).toBe("H1")
+    const { container } = render(VSearchResultsTitle, getOptions())
+    expect(container).toMatchSnapshot()
+  })
+
+  describe("accessible heading", () => {
+    const cs = [0, 10, 4300, 10000, 10001]
+
+    const scenarios = [
```

Comment:
I think there's an opportunity to make this scenario generation a little more readable, at the cost of a more imperative style Here's the general idea and a diff:

```ts
function getScenarios() {
  const counts = [0, 10, 4300, 10000, 10001]
  let scenarios = []
  counts.forEach((count) => {
    counts.forEach((innerCount) => {
      scenarios.push({
        searchType: "all",
        resultCounts: [
          ["image", count],
          ["audio", innerCount],
        ],
      })
    })
    scenarios.push({
      searchType: "image",
      resultCounts: [
        ["image", count],
        ["audio", 0],
      ],
    })
    scenarios.push({
      searchType: "audio",
      resultCounts: [
        ["audio", count],
        ["image", 0],
      ],
    })
  })
  return scenarios.map((scenario) => [JSON.stringify(scenario), scenario])
}
```

And the full diff:

```diff
diff --git a/frontend/test/unit/specs/components/v-search-results-title.spec.js b/frontend/test/unit/specs/components/v-search-results-title.spec.js
index 03c68ce2d..8c4999e56 100644
--- a/frontend/test/unit/specs/components/v-search-results-title.spec.js
+++ b/frontend/test/unit/specs/components/v-search-results-title.spec.js
@@ -12,6 +12,41 @@ const DEFAULT_PROPS = {
   ],
 }
 
+/**
+ * Build test scenarios out of all possible
+ * result count combinations.
+ */
+function getScenarios() {
+  const counts = [0, 10, 4300, 10000, 10001]
+  let scenarios = []
+  counts.forEach((count) => {
+    counts.forEach((innerCount) => {
+      scenarios.push({
+        searchType: "all",
+        resultCounts: [
+          ["image", count],
+          ["audio", innerCount],
+        ],
+      })
+    })
+    scenarios.push({
+      searchType: "image",
+      resultCounts: [
+        ["image", count],
+        ["audio", 0],
+      ],
+    })
+    scenarios.push({
+      searchType: "audio",
+      resultCounts: [
+        ["audio", count],
+        ["image", 0],
+      ],
+    })
+  })
+  return scenarios.map((scenario) => [JSON.stringify(scenario), scenario])
+}
+
 describe("VSearchResultsTitle", () => {
   const getOptions = ({ props, ...options } = {}) => ({
     ...options,
@@ -27,37 +62,7 @@ describe("VSearchResultsTitle", () => {
   })
 
   describe("accessible heading", () => {
-    const cs = [0, 10, 4300, 10000, 10001]
-
-    const scenarios = [
-      ...cs.reduce(
-        (all, i) => [
-          ...all,
-          ...cs.map((a) => ({
-            searchType: "all",
-            resultCounts: [
-              ["image", i],
-              ["audio", a],
-            ],
-          })),
-        ],
-        []
-      ),
-      ...cs.map((count) => ({
-        searchType: "image",
-        resultCounts: [
-          ["image", count],
-          ["audio", 0],
-        ],
-      })),
-      ...cs.map((count) => ({
-        searchType: "audio",
-        resultCounts: [
-          ["audio", count],
-          ["image", 0],
-        ],
-      })),
-    ].map((scenario) => [JSON.stringify(scenario), scenario])
+    let scenarios = getScenarios()
 
     it.each(scenarios)("%s", (_, { searchType, resultCounts }) => {
       const { container } = render(

```